engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/AnalyzeModuleInScopeJob.scala

```diff
@@ -60,8 +61,13 @@ final class AnalyzeModuleInScopeJob(
         exports = ModuleExportsDiff.compute(prevExports, newExports),
         updates = SuggestionDiff.compute(Tree.empty, newSuggestions)
       )
-      sendModuleUpdate(notification)
-      module.setIndexed(true)
+      if (!module.indexed()) {
```

Comment:
Now I see that because it is effectively the only place that marks the module as indexed, just synchronizing around the indexed flag should be enough :+1:

Reply:
> Line 44 is setting an internal module field and leaving synchronized section...

Now we are changing the field again. Meanwhile the state wasn't `synchronized`. What makes you believe the result of such operation is going to be consistent. Btw.

What is the _invariant to be consistent_ with?